### PR TITLE
chore: migrate user namespaces and wipe groups

### PR DESCRIPTION
**Pull request by olevski**
This adds some basic migrations for user namespaces and groups.

/deploy renku=release-0.52.x renku-ui=main

### CODE REVIEW

File: components/renku_data_services/migrations/versions/f34b87ddd954_migrate_namespaces_to_authzed.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            user_w_namespace = UserWithNamespace(user_info, namespace.dump())
+            events = EventConverter.to_events(user_w_namespace, v2.UserAdded)
+            for event in events:
+                execute_coroutine(app_config.event_repo.store_event(session, event))
+
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.drop_index("ix_common_group_members_group_id", table_name="group_members", schema="common")
+    op.drop_table("group_members", schema="common")
+    op.drop_index("ix_users_secrets_user_id", table_name="secrets", schema="secrets")
+    op.create_index(op.f("ix_secrets_secrets_user_id"), "secrets", ["user_id"], unique=False, schema="secrets")
```
[the remark this answers is not in the document]
Comment (olevski) [reply components/renku_data_services/migrations/versions/f34b87ddd954_migrate_namespaces_to_authzed.py:70]: This is a leftover missed migration from secrets. Someone ran the migrations and commited stuff but then changed the ORM and did not regenerate the migrations. So it is a leftover from someone else's PR. We need to add a test that runs `alembic revision --autogenerate` and makes sure the migration is empty on every PR. I see this happening once in a while.

File: components/renku_data_services/migrations/versions/f34b87ddd954_migrate_namespaces_to_authzed.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    authz_client = app_config.authz_config.authz_client()
+    with Session(bind=op.get_bind()) as session, session.begin():
+        # Delete all groups
+        groups = session.scalars(sa.select(GroupORM)).all()
+        for group in groups:
+            session.delete(group)
+            group_model = group.dump()
+            events = EventConverter.to_events(group_model, v2.GroupRemoved)
+            for event in events:
+                execute_coroutine(app_config.event_repo.store_event(session, event))
```
[the remark this answers is not in the document]
Comment (olevski) [reply components/renku_data_services/migrations/versions/f34b87ddd954_migrate_namespaces_to_authzed.py]: Ok I made it be called once not each time every loop.

**Review by Panaetius — APPROVED** [review]
